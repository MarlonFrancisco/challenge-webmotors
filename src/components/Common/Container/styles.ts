import styled from "styled-components";

export const SCContainer = styled.div`
  max-width: ${(props: { maxWidth: number }) => props.maxWidth}px;
  margin: 0 auto;
`;
