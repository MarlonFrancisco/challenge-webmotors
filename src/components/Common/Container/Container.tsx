import React from "react";
import { SCContainer } from "./styles";

enum Breakpoints {
  lg = 933,
  md = 768,
}

export type ContainerProps = {
  maxWidth: "lg" | "md";
};

const Container: React.FC<ContainerProps> = ({ maxWidth, children }) => {
  return <SCContainer maxWidth={Breakpoints[maxWidth]}>{children}</SCContainer>;
};

export default Container;
