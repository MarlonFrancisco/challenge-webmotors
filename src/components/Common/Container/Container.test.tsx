import React from "react";
import { render } from "@testing-library/react";
import Container, { ContainerProps } from "./Container";

const Wrapper = (props: ContainerProps) => <Container {...props} />;

describe("Container", () => {
  let props: ContainerProps & { children: React.ReactNode };

  beforeEach(() => {
    props = {
      maxWidth: "lg",
      children: <p>ContainerComponent</p>,
    };
  });

  it("should render without an error", () => {
    const { container } = render(<Wrapper {...props} />);
    expect(container).toMatchSnapshot();
  });
});
