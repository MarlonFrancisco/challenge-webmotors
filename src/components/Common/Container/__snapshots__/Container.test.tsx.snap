// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render without an error 1`] = `
<div>
  <div
    class="sc-bdnylx iGdfJW"
  >
    <p>
      ContainerComponent
    </p>
  </div>
</div>
`;
